Description: prevent embedding of timestamps in files generated by rcc and uic
 Timestamps in generated files will make packages using those tools no longer
 build reproducibly, because the files will differ in each build.
 As those timestamps are no useful information, they can be left out.
Author: Reiner Herrmann <reiner@reiner-h.de>

Index: python-qt4-4.11.2+dfsg/pyrcc/rcc.cpp
===================================================================
--- python-qt4-4.11.2+dfsg.orig/pyrcc/rcc.cpp
+++ python-qt4-4.11.2+dfsg/pyrcc/rcc.cpp
@@ -24,7 +24,6 @@
 #include <qfile.h>
 #include <qfileinfo.h>
 #include <qlocale.h>
-#include <qdatetime.h>
 #include <qtextstream.h>
 #include <qbytearray.h>
 #include <qhash.h>
@@ -407,8 +406,7 @@ RCCResourceLibrary::writeHeader(FILE *ou
     fprintf(out, "# -*- coding: utf-8 -*-\n\n");
     fprintf(out, "# Resource object code\n");
     fprintf(out, "#\n");
-    fprintf(out, "# Created: %s\n", QDateTime::currentDateTime().toString().toUtf8().constData());
-    fprintf(out, "#      by: The Resource Compiler for PyQt (Qt v%s)\n", QT_VERSION_STR);
+    fprintf(out, "# Created by: The Resource Compiler for PyQt (Qt v%s)\n", QT_VERSION_STR);
     fprintf(out, "#\n");
     fprintf(out, "# WARNING! All changes made in this file will be lost!\n");
     fprintf(out, "\n");
Index: python-qt4-4.11.2+dfsg/pyuic/uic/__init__.py
===================================================================
--- python-qt4-4.11.2+dfsg.orig/pyuic/uic/__init__.py
+++ python-qt4-4.11.2+dfsg/pyuic/uic/__init__.py
@@ -47,8 +47,7 @@ _header = """# -*- coding: utf-8 -*-
 
 # Form implementation generated from reading ui file '%s'
 #
-# Created: %s
-#      by: PyQt4 UI code generator %s
+# Created by: PyQt4 UI code generator %s
 #
 # WARNING! All changes made in this file will be lost!
 
@@ -160,7 +159,6 @@ def compileUi(uifile, pyfile, execute=Fa
     module is foo_rc.
     """
 
-    from time import ctime
     from PyQt4.QtCore import PYQT_VERSION_STR
 
     try:
@@ -170,7 +168,7 @@ def compileUi(uifile, pyfile, execute=Fa
 
     indenter.indentwidth = indent
 
-    pyfile.write(_header % (uifname, ctime(), PYQT_VERSION_STR))
+    pyfile.write(_header % (uifname, PYQT_VERSION_STR))
 
     winfo = compiler.UICompiler().compileUi(uifile, pyfile, from_imports, resource_suffix)
 
